src/cr/cube/matrix.py

```diff
@@ -95,7 +95,31 @@ def table_base(self):
             * CAT_X_CAT - scalar float value when slice has no MR dimension.
 
         """
-        raise NotImplementedError
+        # --- an MR dimension cannot have subtotals, and a non-scalar table base only
+        # --- arises in an MR dimension case, so a non-scalar table-base never has
+        # --- subtotals and we need only operate on the base values. There is no need to
+        # --- assemble with subtotals, however the values of a non-scalar table-base do
+        # --- still need to be ordered (which includes hiding).
+        base_table_base = self._cube_result_matrix.table_base
+
+        # --- MR_X_MR slice produces a 2D table-base (each cell has its own N) ---
+        if (
+            self._rows_dimension.dimension_type == DT.MR_SUBVAR
+            and self._columns_dimension.dimension_type == DT.MR_SUBVAR
+        ):
+            return base_table_base[np.ix_(self._row_order, self._column_order)]
+
+        # --- CAT_X_MR slice produces a 1D table-base (each column has its own N) ---
+        if self._columns_dimension.dimension_type == DT.MR_SUBVAR:
+            return base_table_base[self._column_order]
+
+        # --- MR_X_CAT slice produces a 1D table-base (each row has its own N) ---
+        if self._rows_dimension.dimension_type == DT.MR_SUBVAR:
+            return base_table_base[self._row_order]
+
+        # --- CAT_X_CAT produces scalar table-base ---
+        table_base = self._cube_result_matrix.table_base
```

Comment:
we could return in a single expression, no need for var initialization

Reply:
done